src/Sulu/Bundle/AdminBundle/Navigation/DataNavigationItem.php

```diff
@@ -0,0 +1,267 @@
+    {
+        parent::__construct($name, $parent);
+
+        $this->dataUrl = $dataUrl;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDataUrl()
+    {
+        return $this->dataUrl;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDataResultKey()
+    {
+        return $this->dataResultKey;
+    }
+
+    /**
+     * @param string $dataResultKey
+     */
+    public function setDataResultKey($dataResultKey)
+    {
+        $this->dataResultKey = $dataResultKey;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDataNameKey()
+    {
+        return $this->dataNameKey;
+    }
+
+    /**
+     * @param string $dataNameKey
+     */
+    public function setDataNameKey($dataNameKey)
+    {
+        $this->dataNameKey = $dataNameKey;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDataChildrenLinkKey()
+    {
+        return $this->dataChildrenLinkKey;
+    }
+
+    /**
+     * @param string $dataChildrenLinkKey
+     */
+    public function setDataChildrenLinkKey($dataChildrenLinkKey)
+    {
+        $this->dataChildrenLinkKey = $dataChildrenLinkKey;
+    }
+
+    /**
+     * @return boolean
+     */
+    public function getShowAddButton()
+    {
+        return $this->showAddButton;
+    }
+
+    /**
+     * @param boolean $showAddButton
+     */
+    public function setShowAddButton($showAddButton)
+    {
+        $this->showAddButton = $showAddButton;
+    }
+
+    /**
+     * @return string
+     */
+    public function getInstanceName()
+    {
+        return $this->instanceName;
+    }
+
+    /**
+     * @param string $instanceName
+     */
+    public function setInstanceName($instanceName)
+    {
+        $this->instanceName = $instanceName;
+    }
+
+    /**
+     * @return string
+     */
+    public function getNoDataTranslationKey()
+    {
+        return $this->noDataTranslationKey;
+    }
+
+    /**
+     * @param string $noDataTranslationKey
+     */
+    public function setNoDataTranslationKey($noDataTranslationKey)
+    {
+        $this->noDataTranslationKey = $noDataTranslationKey;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTitleTranslationKey()
+    {
+        return $this->titleTranslationKey;
+    }
+
+    /**
+     * @param string $titleTranslationKey
+     */
+    public function setTitleTranslationKey($titleTranslationKey)
+    {
+        $this->titleTranslationKey = $titleTranslationKey;
+    }
+
+    /**
+     * @param string $addButtonTranslationKey
+     */
+    public function setAddButtonTranslationKey($addButtonTranslationKey)
+    {
+        $this->addButtonTranslationKey = $addButtonTranslationKey;
+    }
+
+    /**
+     * @return string
+     */
+    public function getAddButtonTranslationKey()
+    {
+        return $this->addButtonTranslationKey;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function newInstance()
```

Comment:
Why don't you just use the `clone`-keyword? I guess it could save you all of this code.